cmd/clusterawsadm/ami/helper.go

```diff
@@ -230,17 +234,29 @@ func getAllImages(ec2Client ec2iface.EC2API, ownerID string) (map[string][]*ec2.
 
 func findAMI(imagesMap map[string][]*ec2.Image, baseOS, kubernetesVersion string) (*ec2.Image, error) {
 	amiNameFormat := "capa-ami-{{.BaseOS}}-{{.K8sVersion}}"
-	amiName, err := ec2service.GenerateAmiName(amiNameFormat, baseOS, kubernetesVersion)
+	amiName, err := ec2service.GenerateAmiName(amiNameFormat, baseOS, strings.TrimPrefix(kubernetesVersion, "v"))
```

Comment:
Could we do the trimming of the v in the `GenerateAmiName` function so that it's only done in 1 place?

Reply:
The trimming is not needed for newer AMI names, so took out of GenerateAMIName

Reply:
There is no criteria based on which I could divide it inside `GenerateAMIName`.

Reply:
Reference: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3829/files#diff-a4bd294f7cd314c2947e1a768a7abd5ebd4ed1fa17007b9c9725d1c6d3644926R249

Reply:
We can ignore this. Lets resolve